src/main/java/com/hobbycollection/HobbyCollectionRESTController.java

```diff
@@ -1,15 +1,39 @@
 package com.hobbycollection;
 
+import com.hobbycollection.dao.CollectionDAO;
+import com.hobbycollection.dao.ICollectionDAO;
 import com.hobbycollection.dto.Collection;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.RequestParam;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
 
 @RestController
 public class HobbyCollectionRESTController {
 
+    ICollectionDAO iCollectionDAO;
+
     @GetMapping("/api/Collection/fetchByid")
-    public Collection collectionFetchById(@RequestParam(value="id") int id){
-        return new Collection("Test", "http://something","Test Description",1 );
+        public Collection collectionFetchById(@RequestParam(value="id") int id){
+        try {
+            return new Collection("Test", "http://something", "Test Description", 1);
+        }
+        catch(Exception ex) {
+        ex.printStackTrace();
+            throw ex;
```

Comment:
Add logging here.  Or, better yet, do something to fix the error.
- Catching and immediately rethrowing an error is the same behavior as not having the catch block at all.
- no one will see printStackTrace, and it does not have the concept of priority, as logs do.